app/src/main/java/com/chouten/app/domain/use_case/module_use_cases/AddModuleUseCase.kt

```diff
@@ -338,58 +345,28 @@ class AddModuleUseCase @Inject constructor(
             metadataUriPairs.forEach {
                 log("Comparing module ${module.id} (${module.version}) with ${it.second.id} (${it.second.version})")
                 // Check if the module already exists
+                // TODO: check how the error is handled if the module version is invalid, this was previously done here, now idk
                 if (module.id == it.second.id) {
-                    val ret: Int = try {
-                        module.version.compareSemVer(it.second.version)
-                    } catch (e: IllegalArgumentException) {
-                        // Find which module has the invalid version
-                        try {
-                            "0.0.0".compareSemVer(module.version)
-                        } catch (e: IllegalArgumentException) {
-                            // The module being installed has an invalid version
-                            e.printStackTrace()
-                            safeException(e, newModuleUri)
-                        }
-
-                        try {
-                            "0.0.0".compareSemVer(it.second.version)
-                        } catch (e: IllegalArgumentException) {
-                            e.printStackTrace()
-                            // The existing module has an invalid version
-                            // We can make a note of it and let the module be installed
-                            log("Module ${it.second.name} (${it.second.id}) has an invalid version (${it.second.version})")
-                            1
-                        }
-                    }
-
-                    when (ret) {
-                        // Module being installed is newer than the existing module
-                        1 -> {
-                            // Delete the old module
-                            if (DocumentFile.fromSingleUri(mContext, it.first)?.delete() == false) {
-                                safeException(
-                                    IOException("Could not delete module ${it.second.name} (${it.second.id})"),
-                                    newModuleUri
-                                )
-                            }
-                            log("Updated module ${module.name} (${module.id})")
-                        }
-
-                        // Module being installed is the same version as the existing module
-                        0 -> {
+                    if (module.version < it.second.version) { // old module version < new module version
```

Comment:
Here the "old" module is actually newer. We should remove the old version instead.
```suggestion
                    if (module.version < it.second.version) { // old module version > new module version
                         // Delete the old module
                            if (DocumentFile.fromSingleUri(mContext, newModuleUri)?.delete() == false) {
```